### ISSUE TITLE
Shall we check the return value of ASN1_INTEGER_set() call in crypto/pkcs12/p12_init.c?

### ISSUE THREAD

**Issue by tomgu1991** [openssl/openssl#6569]
[ASN1_INTEGER_set()](https://www.openssl.org/docs/manmaster/man3/ASN1_INTEGER_set.html) sets the value of **ASN1_INTEGER** **a** to the **long** value **v**. The return value 0 is for failure. 

Well, the API reference manual contains a bug
```
long ASN1_INTEGER_set(const ASN1_INTEGER *a);

Shall it be
long ASN1_INTEGER_set(const ASN1_INTEGER *a, long v);

```

In method PKCS12_init() of crypto/pkcs12/p12_init.c, the error check of return value is missed, which may impact the result of PKCS12_init.

Becides, we find the return value of this call been checked in other files in pkcs12 folder.

Such as,
```c
pkcs12/p12_mutl.c,PKCS12_setup_mac:213-220
        if ((p12->mac->iter = ASN1_INTEGER_new()) == NULL) {
            PKCS12err(PKCS12_F_PKCS12_SETUP_MAC, ERR_R_MALLOC_FAILURE);
            return 0;
        }
        if (!ASN1_INTEGER_set(p12->mac->iter, iter)) {
            PKCS12err(PKCS12_F_PKCS12_SETUP_MAC, ERR_R_MALLOC_FAILURE);
            return 0;
        }
```

This problem exists in many versions:
OpenSSL_1_1_0-stable/crypto/pkcs12/p12_init.c,
OpenSSL_1_1_1-pre8/crypto/pkcs12/p12_init.c,
OpenSSL_1_1_1-pre4/crypto/pkcs12/p12_init.c, et al.

Zuxing Gu, Jiecheng Wu

**Comment by kroeckx** [comment]
I fixed the documentation in #6578 (after looking at #6568)

**Comment by mattcaswell** [comment]
#6578 got merged so closing this.
